cpp/include/cugraph_c/sampling_algorithms.h

```diff
@@ -242,6 +309,57 @@ cugraph_error_code_t cugraph_uniform_neighbor_sample_with_edge_properties(
   cugraph_sample_result_t** result,
   cugraph_error_t** error);
 
+/**
+ * @brief     Uniform Neighborhood Sampling
+ *
+ * Returns a sample of the neighborhood around specified start vertices.  Optionally, each
+ * start vertex can be associated with a label, allowing the caller to specify multiple batches
+ * of sampling requests in the same function call - which should improve GPU utilization.
+ *
+ * If label is NULL then all start vertices will be considered part of the same batch and the
+ * return value will not have a label column.
+ *
+ * @param [in]  handle       Handle for accessing resources
+ * @param [in]  graph        Pointer to graph.  NOTE: Graph might be modified if the storage
+ *                           needs to be transposed
+ * @param [in]  start_vertices Device array of start vertices for the sampling
+ * @param [in]  start_vertex_labels  Device array of start vertex labels for the sampling.  The
+ * labels associated with each start vertex will be included in the output associated with results
+ * that were derived from that start vertex.  We only support label of type INT32. If label is
+ * NULL, the return data will not be labeled.
+ * @param [in]  label_list Device array of the labels included in @p start_vertex_labels.  If
+ * @p label_to_comm_rank is not specified this parameter is ignored.  If specified, label_list
```

Comment:
Can you also add that `label_to_comm_rank` cannot be specified  unless `start_vertex_labels` is also specified like it is checked in the test?

Reply:
@ChuckHastings

Reply:
Yes, I'll make that change in #3696

Reply:
Updated in #3696